server/src/main/java/io/crate/role/Role.java

```diff
@@ -37,15 +41,94 @@ public class Role implements ToXContent {
         SUPERUSER
     }
 
+    public record GrantedRole(String roleName, String grantor) implements ToXContent, Writeable {
+
+        public GrantedRole(StreamInput in) throws IOException {
+            this(in.readString(), in.readString());
+        }
+
+        @Override
+        public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+            builder.startObject();
+            builder.field("role", roleName);
+            builder.field("grantor", grantor);
+            builder.endObject();
+
+            return builder;
+        }
+
+        public static GrantedRole fromXContent(XContentParser parser) throws IOException {
+            if (parser.currentToken() != XContentParser.Token.START_OBJECT) {
+                throw new ElasticsearchParseException(
+                    "failed to parse a parent role, expecting the current token to be a start_object, got " +
+                    parser.currentToken()
+                );
+            }
+            String role = null;
+            String grantor = null;
+            while (parser.nextToken() == XContentParser.Token.FIELD_NAME) {
+                switch (parser.currentName()) {
+                    case "role":
+                        parser.nextToken();
+                        role = parser.text();
+                        break;
+                    case "grantor":
+                        parser.nextToken();
+                        grantor = parser.text();
+                        break;
+                    default:
+                        break;
+                }
+            }
+            if (parser.currentToken() != XContentParser.Token.END_OBJECT) {
+                throw new ElasticsearchParseException(
+                    "failed to parse a parent role, expecting the current token to be a end_object, got " +
+                        parser.currentToken()
+                );
+            }
+            return new GrantedRole(role, grantor);
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            out.writeString(roleName);
+            out.writeString(grantor);
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+            GrantedRole that = (GrantedRole) o;
+            return Objects.equals(roleName, that.roleName) && Objects.equals(grantor, that.grantor);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(roleName, grantor);
+        }
+
+        @Override
+        public String toString() {
+            return "ParentRole{" + roleName + ", grantor: " + grantor + '}';
+        }
+    }
+
     private final String name;
     private final boolean isUser;
+    private final Set<GrantedRole> parentRoles;
```

Comment:
```suggestion
    private final Set<GrantedRole> grantedRoles;
```

I'd stick with the terms from the sql statement (grant role to role). The hierarchy implication is imho secondary here and the change of terminology could be interpreted as this being something different or derived